### PR TITLE
Include v7 in IndexVersionUtils#ALL_VERSIONS

### PR DESCRIPTION
**Pull request by javanna**
We are going to retain read-only compatibility with v7 index versions upon upgrade to v9. This means that all the v7 index versions and corresponding version conditionals will stay around for another major series.

This PR reflects this decision in the codebase. No need to filter index versions when retrieving ALL_VERSIONS, and we can remove corresponding @UpdateForV9 annotations from the IndexVersions class. At the same time, there are tests that need to randomize version but need to write to an index, hence 7x versions should be filtered out.

The overall goal is to extend testing to v7 index versions when possible, and making randomization across writeable versions an exception that certain tests can rely on as needed.

These are the mechanical steps I made in this PR:

- Rename `getFirstVersion` to `getLowestReadCompatibleVersion`, which returns the lowest supported index version ( which can not be written to)
- Introduce `getLowestWriteCompatibleVersion` to identify the lowest writeable version. This is used by tests that used to call `getFirstVersion` and need to write to the index.
- Remove `randomVersion(Random random)` in favour of `randomVersion()` . It was always called providing `random()` which is equivalent to what `randomVersion()` already does.
- Introduce `randomWriteVersion` for tests that need a random writeable version. Moved tests that need it to use it (from `randomVersion` to `randomWriteVersion`)

There is still work to do in `IndexVersionUtils` to extend testing, especially as some randomized tests have a lower bound of `MINIMUM_COMPATIBLE_VERSION` hence don't include v7 index versions yet. We will address that as a follow-up.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexServiceTests.java [hunk 1]
```diff
@@ -369,11 +369,7 @@ public void testValidateSplitIndex() {
     }
 
     public void testPrepareResizeIndexSettings() {
-        final List<IndexVersion> versions = Stream.of(IndexVersionUtils.randomVersion(random()), IndexVersionUtils.randomVersion(random()))
-            .sorted()
-            .toList();
-        final IndexVersion version = versions.get(0);
-        final IndexVersion upgraded = versions.get(1);
```
Comment (javanna) [inline server/src/test/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexServiceTests.java:376]: The second version here was unused, so I removed the unused bits

File: server/src/test/java/org/elasticsearch/index/IndexVersionTests.java [hunk 2]
```diff
@@ -151,9 +150,7 @@ public void testMax() {
         }
     }
 
-    @UpdateForV9(owner = UpdateForV9.Owner.CORE_INFRA)
-    @AwaitsFix(bugUrl = "believe this fails because index version has not yet been bumped to 9.0")
-    public void testMinimumCompatibleVersion() {
```
Comment (javanna) [inline server/src/test/java/org/elasticsearch/index/IndexVersionTests.java:156]: This was an outdated awaitsfix, we can simply remove it. I went ahead and renamed the test method as well for clarity

File: server/src/test/java/org/elasticsearch/index/IndexVersionTests.java [hunk 3]
```diff
@@ -193,8 +190,6 @@ public void testParseLenient() {
         }
     }
 
-    @UpdateForV9(owner = UpdateForV9.Owner.CORE_INFRA)
-    @AwaitsFix(bugUrl = "can be unmuted once lucene is bumped to version 10")
```
Comment (javanna) [inline server/src/test/java/org/elasticsearch/index/IndexVersionTests.java:197]: No update needed given that we still support reading from v7 indices, hence Lucene 8 is a known major version (what `getLowestReadCompatibleVersion.luceneVersion().major` returns).

**Comment by javanna** [comment]
Marking Core/Infra as well because I removed some of the `@UpdateForV9` with core/infra set as an owner. Yet they were all in the index versions area so the owner could have rather been Search Foundations.

**Review by cbuescher — APPROVED** [review]
LGTM

**Review by drempapis — APPROVED** [review]
Thank you @javanna. LGTM, despite its size, has only introduced a few things.

**Review by alexey-ivanov-es — APPROVED** [review]
LGTM, thank you for taking care of those Core/Infra's @UpdateForV9s

**Comment by javanna** [comment]
Thanks all for the reviews!

**Pull request merged**